src/eth/index.ts

```diff
@@ -76,14 +88,14 @@ class Eth {
         });
       }
 
-      const payload = {
+      const payload: RPCPayload<P> = {
         call,
         params,
       };
 
-      const message = this.communicator.send(SDK_MESSAGES.RPC_CALL, payload, args.requestId);
+      const response = await this.#communicator.send<'rpcCall', RPCPayload<P>, R>(METHODS.rpcCall, payload);
```

Comment:
How do we handle errors here? (e.g. if the rpc calls fails)

Reply:
It will return an error response

Reply:
but the `buildRequest` is called with type (`R`) which are not compliant to an error ... so how would this work type wise?

Reply:
oops 😛 I think it's better to actually have a check inside the `buildRequest` function and throw if it returns an error. Good catch